### PR TITLE
Speed up map loading for New Game

### PR DESCRIPTION
**Pull request by ihhub**
A small performance improvement.

### CODE REVIEW

File: src/fheroes2/game/game_scenarioinfo.cpp [hunk 1]
```diff
@@ -394,22 +394,19 @@ namespace
...
-
-        if ( lower.size() > 3 ) {
-            std::string ext = lower.substr( lower.size() - 3 );
+        const std::string & filePath = conf.MapsFile();
+        const size_t pos = filePath.rfind( '.' );
+        if ( pos != std::string::npos ) {
+            const std::string ext = StringLower( filePath.substr( pos + 1 ) );
 
             if ( ext == "mp2" || ext == "mx2" ) {
-                return world.LoadMapMP2( conf.MapsFile(), ( ext == "mp2" ) ) ? fheroes2::GameMode::START_GAME : fheroes2::GameMode::MAIN_MENU;
```
Comment (Districh-ru) [inline src/fheroes2/game/game_scenarioinfo.cpp:404]: What do you think if we also cache: `const bool isSwMap = ( ext == "mp2" );` ?

**Review by Districh-ru — APPROVED** [review]
Hi, @ihhub, well done!
I left one optional comment with minor optimization suggestion. You can address it if you think this suggestion is reasonable.

File: src/fheroes2/game/game_scenarioinfo.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/game/game_scenarioinfo.cpp:404]: Hi @Districh-ru , this will make the code less readable with the given improvement.

Comment (Districh-ru) [reply src/fheroes2/game/game_scenarioinfo.cpp:404]: @ihhub,
OK

**Pull request merged**